pkgs/by-name/sp/speech-provider-piper/package.nix

```diff
@@ -0,0 +1,73 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cargo,
+  meson,
+  ninja,
+  pkg-config,
+  rustPlatform,
+  rustc,
+  glib,
+  onnxruntime,
+  darwin,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "speech-provider-piper";
+  version = "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = "project-spiel";
+    repo = "speech-provider-piper";
+    rev = "SPEECH_PROVIDER_PIPER_${lib.replaceStrings [ "." ] [ "_" ] finalAttrs.version}";
+    hash = "sha256-vkKZ75yJeoAditIFZ0RvsCF/chDYzJwqC4WpdZTHEyM=";
+  };
+
+  cargoDeps = rustPlatform.importCargoLock {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "audio-ops-1.0.0" = "sha256-P27R4S3HwI1FJJsSLc0q+j082MTpVmNrf8JuAhPsUxo=";
+      "glib-sys-0.20.0" = "sha256-ICNIuE3/6dLBBOTbIW9DjaEg1+tCmHZmgQw/zv/1bIU=";
+      "speechprovider-0.1.0" = "sha256-Z+Fzec5uE87SfHCt5h/Hnts2RJ08twgm13HDvgLRiZ0=";
+    };
+  };
+
+  nativeBuildInputs = [
+    cargo
+    meson
+    ninja
+    pkg-config
+    rustPlatform.bindgenHook
+    rustPlatform.cargoSetupHook
+    rustc
+  ];
+
+  buildInputs =
+    [
+      glib
+      onnxruntime
+    ]
+    ++ lib.optionals stdenv.isDarwin [
+      darwin.apple_sdk.frameworks.Security
+    ];
+
+  strictDeps = true;
+
+  mesonFlags = [
+    "-Doffline=true"
+  ];
+
+  env = {
+    ORT_PREFER_DYNAMIC_LINK = "1";
```

Comment:
with these I got ORT to build:

```
    ORT_STRATEGY = "system";
    ORT_LIB_LOCATION = "${onnxruntime}/lib";
    OPENSSL_NO_VENDOR = true;
```

Reply:
inspired by https://github.com/NixOS/nixpkgs/pull/364392/files#diff-976605ef57cfc705a3d9475324ea397610994350dadc9e3d79d6cb82cd2fee08R72-R75